processing/WordVectors.py

```diff
@@ -0,0 +1,126 @@
+import json
+import random
+import re
+import sys
+
+import matplotlib.pyplot as plt
+import numpy
+import spacy
+from gensim.parsing.preprocessing import preprocess_string, remove_stopwords
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+
+
+def get_words(text):
+    """
+    Source: <@FIXME:/somewhere/>
+    """
+    text = text.lower()
+    wordlist = text.split()
+    clean_list = []
+    for word in wordlist:
+        # only get words (no digits)
+        if not word.isdigit() and not re.match(r"[^\w]", word):
+            clean_list.append(word)
+    return " ".join(clean_list)
+
+
+class WordVectors:
+    """
+    @FIXME: class to represent word vectors. more stuff here
+    """
+
+    dataset = []  # list of dictionaries with "summary" and "_id" fields
+    # @FIXME: or perhaps we ask for a more general input format?
+    samples = 500  # just used for visualization; we don't want to dump all points
+    # onto the plot, as they will overlap and overwhelm
+
+    word_vectors = []
+    word_embedding = None  # word embedding model; pretrained from spacy.io
+
+    X = None
+    explained_variance = None
+
+    def __init__(self, data_file: string, samples: int):
```

Comment:
this won't work; also change to data in memory input. samples should have a default value